spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/ZuulProxyConfiguration.java

```diff
@@ -88,7 +88,9 @@ public DiscoveryClientRouteLocator routeLocator() {
 	public PreDecorationFilter preDecorationFilter(RouteLocator routeLocator) {
 		return new PreDecorationFilter(routeLocator,
 				this.zuulProperties.isAddProxyHeaders(),
-				this.zuulProperties.isRemoveSemicolonContent());
+				this.zuulProperties.isRemoveSemicolonContent(),
+				this.server.getServletPrefix(),
+				this.zuulProperties);
```

Comment:
I reckon, pass the `ZuulProperties` if it avoids a tangle. If not create a new DTO and initialize it from `ZuulProperties` before you pass it in.